internal/overlord/servstate/manager.go

```diff
@@ -300,6 +300,24 @@ func (m *ServiceManager) Services(names []string) ([]*ServiceInfo, error) {
 	return services, nil
 }
 
+// MaximumStopRunningDuration returns the worst case duration that will have to
+// be waited for to have all services in this manager stopped.
+func (m *ServiceManager) MaximumStopRunningDuration() time.Duration {
+	m.servicesLock.Lock()
+	defer m.servicesLock.Unlock()
+
+	maxDuration := defaultKillDelay
+	for _, svc := range m.services {
+		if svc.killDelay() > maxDuration {
```

Comment:
A few comments here:

1. Shouldn't we be looking only at `s.state == stateRunning` here?
2. Other code that look into the map seem to consider the case where the map value is nil, which I assume means a quick shutdown. Do we have a panic waiting to happen here?
3. The code in this file (and in tests?) doesn't seem to use the `svc` naming. It's nice to keep these consistent when possible (or more consistent, if there are already multiple cases).

Reply:
Thanks, I have re-worked the code a little and updated tests.

I am considering starting, running and terminating.

Terminating might seem like an odd choice my justification is if you have service a that has a kill delay of 5 minutes and you tell it to stop then immediately stop Pebble the next wait time will be what ever the maximum is of the running services we still might fall outside of that 5 minute window.

Hope that makes sense.